### ISSUE TITLE
State machines 🎰

### ISSUE THREAD

**Issue by bouwe77** [bouwe77/curious-coders#32]
How to write your logic as a state machine, and why it is cool, and why maybe even a good idea...

**Comment by RichStone** [comment]
yeah, man, we have quite a bit of state machinery in the back too, but I haven't dove deeper it yet (until now obviously :))
